test/k8sT/Policies.go

```diff
@@ -1442,14 +1442,16 @@ var _ = Describe("K8sPolicyTest", func() {
 				importPolicy(kubectl, testNamespace, cnpDenyIngress, "default-deny-ingress")
 
 				count := testConnectivity(backendPodIP, false)
-				monitorCancel()
+				defer monitorCancel()
 
 				By("Asserting that the expected policy verdict logs are in the monitor output")
-				Expect(
-					len(policyVerdictDenyRegex.FindAll(monitor.CombineOutput().Bytes(), -1)),
-				).To(BeNumerically(">=", count),
-					"Monitor output does not show traffic as denied: %s\n%s", policyVerdictDenyRegex,
-					monitor.CombineOutput().Bytes())
+				var verdicts int
+				err = helpers.RepeatUntilTrueDefaultTimeout(func() bool {
+					verdicts = len(policyVerdictDenyRegex.FindAll(monitor.CombineOutput().Bytes(), -1))
+					return verdicts >= count
+				})
+				Expect(err).To(BeNil(), "Monitor output is missing verdicts (%d < %d): %s\n%s",
+					verdicts, count, policyVerdictDenyRegex, monitor.CombineOutput().Bytes())
```

Comment:
I think we should use `Eventually` here as that's the idiom for Ginkgo. It also a more expressive way to assert that the verdict is what we expect. 

```suggestion
				Eventually(func() int {
					return len(policyVerdictDenyRegex.FindAll(monitor.CombineOutput().Bytes(), -1))
				}, DefaultTimeout).Should(BeNumerically(">=", count))
```

Reply:
Ah, nice! Does that mean we should just remove our `RepeatUntil` helpers?

Reply:
I think so. I'm not sure why we need them if Ginkgo has native support for polling a condition. They were likely added because others were unaware of the Ginkgo functionality.

Reply:
I pushed a new commit. I've not removed the functions after all because there are a couple of legitimate uses (AFAICT).